### PR TITLE
Chore/4.0/update api gen for new api

### PR DESCRIPTION
**Pull request by piiertho**
This updates api gen for the beta10 godot 4.0.  

There's some issue with godot api, which involve workarounds on our side:

https://github.com/godotengine/godot/issues/70734
https://github.com/godotengine/godot/issues/70736

### CODE REVIEW

File: kt/api-generator/src/main/kotlin/godot/codegen/workarounds/SignalWorkarounds.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package godot.codegen.workarounds
+
+import godot.codegen.models.enriched.EnrichedSignal
+
+@Deprecated(
+    "This should be deleted when godot api is clean",
+    replaceWith = ReplaceWith("EnrichedSignal.name")
+)
+fun EnrichedSignal.sanitizedName(englobingClassName: String) =
```
Comment (CedNaru) [inline kt/api-generator/src/main/kotlin/godot/codegen/workarounds/SignalWorkarounds.kt:9]: Put also a Todo here with the Godot Issue

File: kt/api-generator/src/main/kotlin/godot/codegen/extensions/TypedExtensions.kt [hunk 2]
```diff
@@ -8,12 +8,16 @@ import godot.codegen.traits.NullableTrait
...
-fun TypedTrait.isCoreType() = GodotTypes.coreTypes.find { s -> s == this.type } != null
+private const val enumPrefix = "enum::"
+
+fun TypedTrait.isCoreType() = isTypedArray() || GodotTypes.coreTypes.find { s -> s == this.type } != null
 fun TypedTrait.isPrimitive() = GodotTypes.primitives.find { s -> s == this.type } != null
 fun TypedTrait.isCoreTypeReimplementedInKotlin() = GodotTypes.coreTypesReimplementedInKotlin.find { s ->
     s == this.type
 } != null
-fun TypedTrait.isEnum() = type?.startsWith("enum::") ?: false
+fun TypedTrait.isEnum() = type?.startsWith(enumPrefix) ?: false
```
[the remark this answers is not in the document]
Comment (piiertho) [reply kt/api-generator/src/main/kotlin/godot/codegen/extensions/TypedExtensions.kt:19]: Should be fixed

File: kt/api-generator/src/main/kotlin/godot/codegen/extensions/StringExtensions.kt [hunk 3]
```diff
@@ -39,6 +39,8 @@ fun String.convertToSnakeCase(): String =
 
 fun String.escapeKotlinReservedNames() = if (kotlinReservedNames.find { s -> s == this } != null) "_$this" else this
 
+fun String.sanitizeApiType() = this.replace("24/17:", "")
```
[the remark this answers is not in the document]
Comment (piiertho) [reply kt/api-generator/src/main/kotlin/godot/codegen/extensions/StringExtensions.kt]: Should be fixed

File: kt/api-generator/src/main/kotlin/godot/codegen/workarounds/SignalWorkarounds.kt (hunk 1 above)
Comment (piiertho) [reply kt/api-generator/src/main/kotlin/godot/codegen/workarounds/SignalWorkarounds.kt:11]: Should be fixed

**Review by CedNaru — APPROVED** [review]

**Pull request merged**